habitat-lab/habitat/datasets/rearrange/samplers/receptacle.py

```diff
@@ -505,34 +499,85 @@ def get_all_scenedataset_receptacles(
     return receptacles
 
 
-def import_tri_mesh_ply(ply_file: str) -> TriangleMeshData:
+def filter_interleave_mesh(mesh: mn.trade.MeshData) -> mn.trade.MeshData:
     """
-    Returns a Tuple of (verts,indices) from a ply mesh using magnum trade importer.
+    Filter all but position data and interleave a mesh to reduce overall memory footprint.
+    Convert triangle like primitives into triangles and assert only triangles remain.
 
-    :param ply_file: The input PLY mesh file. NOTE: must contain only triangles.
+    NOTE: Modifies the mesh data in-place
+    :return: The modified mesh for easy of use.
     """
-    manager = mn.trade.ImporterManager()
-    importer = manager.load_and_instantiate("AnySceneImporter")
-    importer.open_file(ply_file)
-
-    # TODO: We don't support mesh merging or multi-mesh parsing currently
-    if importer.mesh_count > 1:
-        raise NotImplementedError(
-            "Importing multi-mesh receptacles (mesh merging or multi-mesh parsing) is not supported."
-        )
 
-    mesh_ix = 0
-    mesh = importer.mesh(mesh_ix)
+    # convert to triangles and validate the result
+    if mesh.primitive in [
+        mn.MeshPrimitive.TRIANGLE_STRIP,
+        mn.MeshPrimitive.TRIANGLE_FAN,
+    ]:
+        mesh = mn.meshtools.generate_indices(mesh)
     assert (
         mesh.primitive == mn.MeshPrimitive.TRIANGLES
     ), "Must be a triangle mesh."
 
-    # zero-copy reference to importer datastructures
-    mesh_data = TriangleMeshData(
-        mesh.attribute(mn.trade.MeshAttribute.POSITION),
-        mesh.indices,
+    # filter out all but positions (and indices) from the mesh
+    mesh = mn.meshtools.filter_only_attributes(
+        mesh, [mn.trade.MeshAttribute.POSITION]
     )
 
+    # reformat the mesh data after filtering
+    mesh = mn.meshtools.interleave(mesh, mn.meshtools.InterleaveFlags.NONE)
+
+    return mesh
+
+
+def import_tri_mesh(mesh_file: str) -> List[mn.trade.MeshData]:
+    """
+    Returns a list of MeshData objects from a mesh asset using magnum trade importer.
+
+    :param mesh_file: The input meshes file. NOTE: must contain only triangles.
+    """
+    _importer = _manager.load_and_instantiate("AnySceneImporter")
```

Comment:
```suggestion
    importer = _manager.load_and_instantiate("AnySceneImporter")
```

This probably counts as trolling now, sorry.